### PR TITLE
George2

### PR DESCRIPTION
**Pull request by get13**
Interface functional, interacts with Parse correctly. 
Needs shipment calculations from previous User Stories

### CODE REVIEW

File: src/main/java/coffeeimport/InventoryController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Created by George Totolos on 7/6/2015.
+ */
+@Controller
+@RequestMapping("/inventory")
+
+public class InventoryController
+{
+
```
Comment (AdamRison) [inline src/main/java/coffeeimport/InventoryController.java]: You should add a minor description of what this class is for or what the method does.

File: src/main/java/coffeeimport/NewProductController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class NewProductController
+{
+
```
Comment (AdamRison) [inline src/main/java/coffeeimport/NewProductController.java]: I know these aren't big methods, but you should still add a comment or a JDoc with a small description of what this is for.  Just enough of a blurb so that someone looking at this can read it and say, "Ohh, this method does ______."

File: src/main/java/coffeeimport/NewProductController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    @RequestMapping(value="/newproduct",method = RequestMethod.GET)
+    public String beginProduct(Model model)
+    {
+        model.addAttribute("product", new Product());
+        return "newShipment";
+    }
+
```
Comment (AdamRison) [inline src/main/java/coffeeimport/NewProductController.java]: A comment or JDoc would be good for this method too.

File: src/main/java/coffeeimport/ParseAccessor.java [hunk 4]
```diff
@@ -40,8 +40,11 @@ public ParseAccessor(String id, String key){
     public ParseAccessor(){
         //Need to assume an id and key;  Using mine (Adam's) until otherwise specified
         Parse.initialize("lXpYJrImjyl3YSKDvxX9R6H3GGqIKQrB6WbI6Eu1", "dRwFZFBIxUiHP8nL3JYKsjtJpCLX1SZCl7Zez5C3");
```
Comment (AdamRison) [inline src/main/java/coffeeimport/ParseAccessor.java]: You should leave the ID and Key out before it goes to the Master branch, just to be safe.

File: src/main/java/coffeeimport/ShipmentController.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Created by George Totolos on 7/6/2015.
+ */
+@Controller
+@RequestMapping("/inventory")
+
```
Comment (AdamRison) [inline src/main/java/coffeeimport/ShipmentController.java]: Maybe add a brief comment on what this is.

File: src/main/java/resources/templates/inventory.html [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+    <link rel="stylesheet" href="//code.jquery.com/ui/1.11.4/themes/smoothness/jquery-ui.css">
+    <script src="//code.jquery.com/ui/1.11.4/jquery-ui.js"></script>
+
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+
+</head>
+<body>
+<script>
+    // Access Parse database (ID, key)
+    Parse.initialize("lXpYJrImjyl3YSKDvxX9R6H3GGqIKQrB6WbI6Eu1", "tmfM1mnCsFEE9onzPSAqwKubvgKakqPppNx8dnsq");
```
Comment (AdamRison) [inline src/main/java/resources/templates/inventory.html]: You should leave the ID and Key out before it goes to the Master branch, just to be safe.

File: src/main/java/resources/templates/inventory.html [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+                    <p><input id="parse-submit" type="submit"></p>
+                </form>
+            </div>
+        </div>
+
+        <!----------------------------------- Modify Product ----------------------------------->
+        <!--
+        USAGE:
+            User input ProductID field then presses "Find Product" button. Other field auto-generate
+            based on parse entry. From all the user has to do is change whatever fields they want and
```
Comment (AdamRison) [inline src/main/java/resources/templates/inventory.html]: I think this is a typo, "From all the user has to do".  I like that you've commented what this code does though.

File: src/main/java/resources/templates/main.html [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    <link rel="stylesheet" href="styles/style.css">
+    <script src="//www.parsecdn.com/js/parse-1.4.2.min.js"></script>
+    <script src="http://code.jquery.com/jquery-1.9.1.min.js"></script>
+    <script src="scripts/scripts.js"></script>          <!-- External javascript/jquery to handle tabbing -->
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+</head>
+<body>
+    <script>
+        // Access Parse database (ID, key)
+        Parse.initialize("lXpYJrImjyl3YSKDvxX9R6H3GGqIKQrB6WbI6Eu1", "tmfM1mnCsFEE9onzPSAqwKubvgKakqPppNx8dnsq");
```
Comment (AdamRison) [inline src/main/java/resources/templates/main.html]: Here too, you should take out the ID and key to be safe.

File: src/main/java/resources/templates/scripts/productQuery.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                cell6.innerHTML = sold;
+                cell7.innerHTML = priceSold;
+                cell8.appendChild(deleteBtn);
+
+
+
+                rowCount = rowCount + 1;
+
+            }
+        }, error: function (error) {
```
Comment (AdamRison) [inline src/main/java/resources/templates/scripts/productQuery.js]: You might want to comment on what would cause this error case.

File: src/main/java/resources/templates/scripts/shipmentQuery.js [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+                var cell6 = row.insertCell(5);
+                var cell7 = row.insertCell(6);
+                var cell8 = row.insertCell(7);
+                var cell9 = row.insertCell(8);
+                //var cell10 = row.insertCell(9);
+
+
+/*
+
+                // Create Edit button to be generated for each row
```
Comment (AdamRison) [inline src/main/java/resources/templates/scripts/shipmentQuery.js]: If you aren't using this code anymore, does it need to be here?

File: src/main/java/resources/templates/scripts/shipmentQuery.js [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+            }
+        }, error: function (error) { // Error loading query from parse
+            alert("Could not retrieve Shipment data");
+        }
+    });
+}
+
+
+/*
+function removeProduct(prodID) { // Removes product based on user entered ID
```
Comment (AdamRison) [inline src/main/java/resources/templates/scripts/shipmentQuery.js]: Same for the following code.  If you aren't going to use it, might as well cut it out and free up the space.

File: src/main/java/resources/templates/shipments.html [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+    <link rel="stylesheet" href="//code.jquery.com/ui/1.11.4/themes/smoothness/jquery-ui.css">
+    <script src="//code.jquery.com/ui/1.11.4/jquery-ui.js"></script>
+
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+
+</head>
+<body>
+<script>
+    // Access Parse database (ID, key)
+    Parse.initialize("lXpYJrImjyl3YSKDvxX9R6H3GGqIKQrB6WbI6Eu1", "tmfM1mnCsFEE9onzPSAqwKubvgKakqPppNx8dnsq");
```
Comment (AdamRison) [inline src/main/java/resources/templates/shipments.html]: Here, you should take out the ID and key for safety.

File: src/main/java/resources/templates/shipments.html [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+    <script>
+        // Generate table through Parse Query
+        getShipmentsTable();
+    </script>
+
+    <!----------------------------------- Modify Shipment ----------------------------------->
+    <!--
+    USAGE:
+        User input ShipmentID field then presses "Find Shipment" button. Other field auto-generate
+        based on parse entry. From all the user has to do is change whatever fields they want and
```
Comment (AdamRison) [inline src/main/java/resources/templates/shipments.html]: Just noticed the same typo "From all the user has to."  I know you said that we're probably going to focus on this file as opposed to the inventory.html file.  If that's the case, do we need the inventory file anymore, and if we don't, can the program function without it?

File: src/main/java/resources/templates/shipments.html [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+                alert("Entered product cost is not a valid number");
+                return;
+            }
+
+            if ( isNaN(priceKG) ){ // if price per kg is invalid
+                alert("Entered price per kg is not a valid number");
+                return;
+            }
+
+            // handle NULL values
```
Comment (AdamRison) [inline src/main/java/resources/templates/shipments.html]: Glad you found a way to fix the problem.  Did you test to see if it works for each case?

File: src/test/java/coffeeimport/GeorgesParseAccessorTests.java [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+        testProduct.setProductName("George's Coffee");
+        testProduct.setProductCost(100);
+        testProduct.setHasBeenSold(false);
+        testProduct.setPriceSold(150);
+
+
+        parseAccessor.updateProduct(testProduct);
+*/
+    }
+
```
Comment (AdamRison) [inline src/test/java/coffeeimport/GeorgesParseAccessorTests.java]: Considering this file doesn't actually do anything or test any code, why was it added?  Also, if it needs to stay, for whatever reason, you should remove the Parse ID and key for safety.

File: src/test/java/coffeeimport/ParseAccessorTests.java [hunk 16]
```diff
@@ -23,6 +24,7 @@
      */
     @Before
     public void instantiate() throws Exception{
+        // Adam's ID:
         parseAccessor = new ParseAccessor("lXpYJrImjyl3YSKDvxX9R6H3GGqIKQrB6WbI6Eu1", "dRwFZFBIxUiHP8nL3JYKsjtJpCLX1SZCl7Zez5C3");
```
Comment (AdamRison) [inline src/test/java/coffeeimport/ParseAccessorTests.java]: You should remove the Parse ID and key here too.

**Comment by AdamRison** [comment]
At a glance, it looks like you added a lot of stuff and fixed some of the problems you were talking about, so good.  You should add more comments and remove the Parse IDs and Keys where you have them so others can't get access to them.  Also, did you test to make sure that everything doesn't break?  I saw that you added code to handle null values, but does it work?
